cluster/src/main/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializer.java

```diff
@@ -593,75 +815,385 @@ public void close() {
   }
 
   /**
-   * adjust maxRemovedLogSize to the first log file
+   * get file version from file The file name structure is as follows：
+   * {startLogIndex}-{endLogIndex}-{version}-data)
+   *
+   * @param file file
+   * @return version from file
+   */
+  private long getFileVersion(File file) {
+    return Long.parseLong(file.getName().split(FILE_NAME_SEPARATOR)[2]);
+  }
+
+  public void checkDeletePersistRaftLog() {
+    // 1. check the log index offset list size
+    try {
+      lock.writeLock().lock();
+      if (logIndexOffsetList.size() > maxRaftLogIndexSizeInMemory) {
+        int compactIndex = logIndexOffsetList.size() - maxRaftLogIndexSizeInMemory;
+        logIndexOffsetList.subList(0, compactIndex).clear();
+        firstLogIndex += compactIndex;
+      }
+    } finally {
+      lock.writeLock().unlock();
+    }
+
+    // 2. check the persist log file number
+    while (logDataFileList.size() > maxNumberOfPersistRaftLogFiles) {
+      deleteLogDataAndIndexFile(0);
+    }
+
+    // 3. check the persist log index number
+    while (!logDataFileList.isEmpty()) {
+      File firstFile = logDataFileList.get(0);
+      String[] splits = firstFile.getName().split(FILE_NAME_SEPARATOR);
+      if (meta.getCommitLogIndex() - Long.parseLong(splits[1]) > maxPersistRaftLogNumberOnDisk) {
+        deleteLogDataAndIndexFile(0);
+      } else {
+        return;
+      }
+    }
```

Comment:
If `maxPersistRaftLogNumberOnDisk` is too small, is it possible that the current file will be deleted here?

Reply:
Thank you for reminding me that this could happen. I'll fix it